### PR TITLE
Switch to GitHub Actions and upgrade dev tooling

### PR DESCRIPTION
**Pull request by ixti**
- Update rubocop
- Switch to GitHub Actions
- Update copyrights years

I think API needs to be significantly rewritten:
- File should not inherit from Part, which will be a breaking change.
- It should use kwargs instead of opts hash

### CODE REVIEW

**Comment by ixti** [comment]
This change is [Reviewable](https://reviewable.io/reviews/httprb/form_data/31)

File: lib/http/form_data.rb [hunk 1]
```diff
@@ -57,10 +57,9 @@ def create(data, encoder: nil)
       # @raise [Error] `obj` can't be coerced.
       # @return [Hash]
       def ensure_hash(obj)
-        case
-        when obj.nil?               then {}
-        when obj.is_a?(Hash)        then obj
-        when obj.respond_to?(:to_h) then obj.to_h
+        if    obj.nil?               then {}
+        elsif obj.is_a?(Hash)        then obj
+        elsif obj.respond_to?(:to_h) then obj.to_h
```
Comment (freemanoid) [inline lib/http/form_data.rb:62]: This doesn't look prettier. I wonder what's the reason of this change?

Comment (ixti) [reply lib/http/form_data.rb:62]: Rubocop :D

Comment (ixti) [reply lib/http/form_data.rb:62]: I guess it doesn't look much worse either... :D
